MapboxNavigation/NavigationViewController.swift

```diff
@@ -589,18 +591,22 @@ extension NavigationViewController: NavigationServiceDelegate {
             currentLeg = progress.currentLeg
             passedApproachingDestinationThreshold = false
             mapViewController?.supressAutomaticAltitudeChanges = false
+            foundAllBuildings = false
             if let mapView = mapView {
                 mapView.altitude = mapView.defaultAltitude
             }
         }
         
-        if let mapView = mapView, passedApproachingDestinationThreshold == false, waypointStyle != .annotation, let currentLegWaypoint = progress.currentLeg.destination?.targetCoordinate, progress.currentLegProgress.distanceRemaining < approachingDestinationThreshold {
+        if let mapView = mapView, passedApproachingDestinationThreshold == false, progress.currentLegProgress.distanceRemaining < approachingDestinationThreshold {
             passedApproachingDestinationThreshold = true
             mapViewController?.supressAutomaticAltitudeChanges = true
```

Comment:
Do we still want to set this value to true even if the waypoint style is `annotation`?